### PR TITLE
build: add mkdocs for documentation generation (DSP-460)

### PR DESCRIPTION
**Pull request by subotic**
- [x] add MkDocs site configuration
- [x] update Makefiles
- [x] update documentation on how to use it
- [x] publish to docs.knora.org
- [x] update links on knora.org
- [x] automate publishing docs on every release

=> orchestration through Bazel will be implemented in a future PR ([DSP-469](https://dasch.myjetbrains.com/youtrack/issue/DSP-469))

### CODE REVIEW

File: docs/05-internals/design/api-v2/json-ld.md [hunk 1]
```diff
@@ -19,8 +19,6 @@ License along with Knora.  If not, see <http://www.gnu.org/licenses/>.
 
 # JSON-LD Parsing and Formatting
```
Comment (benjamingeer) [inline docs/05-internals/design/api-v2/json-ld.md:20]: Can you fix the code snippets? For example, in this file, there are several `@@snip` directives, which refer to snippets that are marked with special `// #` comments in the source code (e.g. in `ValuesRouteV2.scala`. These comments need to be deleted, and the snippets need to be copied and pasted into the documentation.

You can find all the Markdown files containing code snippets by searching for `snip` on docs.knora.org. :)

Comment (subotic) [reply docs/05-internals/design/api-v2/json-ld.md:20]: ok, I've copied everything. no `@@snip` left.

File: docs/Readme.md [hunk 2]
```diff
@@ -1,56 +1,28 @@
 # Knora Documentation
 
-## 1 Paradox Documentation
+## 1 MkDocs Documentation
```
Comment (benjamingeer) [inline docs/Readme.md:3]: This file still talks about Paradox and Jekyll, shouldn't that part be deleted? Also, the sections are numbered 1, 4, 5, 6.

Comment (subotic) [reply docs/Readme.md]: Thanks. Now there should be no more mention of paradox including any links.

**Comment by subotic** [comment]
Thanks for the review :-)

**Pull request merged**